command/init.go

```diff
@@ -519,6 +519,18 @@ func (c *InitCommand) getProviders(earlyConfig *earlyconfig.Config, state *state
 
 			c.Ui.Info(fmt.Sprintf("- Installed %s v%s (%s)%s", provider.ForDisplay(), version, authResult, warning))
 		},
+		ProvidersFetched: func(authResults map[addrs.Provider]*getproviders.PackageAuthenticationResult) {
+			thirdPartySigned := false
+			for _, authResult := range authResults {
+				if authResult.ThirdPartySigned() {
+					thirdPartySigned = true
+					break
+				}
+			}
+			if thirdPartySigned {
+				c.Ui.Info(fmt.Sprintf("\nYou have initialized providers which are signed by third parties, you can read more about Terraform's signature verification: https://www.terraform.io/docs/signing.html"))
```

Comment:
In discussing this a little more with the team, I've got a suggestion to move this from warning to informational in tone. This is a suggestion:

```
Partner and community providers are now signed by their developers. 
If you'd like to know more about provider signing, you can read about 
it here: https://www.terraform.io/docs/signing.html
```

Reply:
I like that!

Reply:
I really like this language improvement.